src/client/components/three/computed.ts

```diff
@@ -0,0 +1,198 @@
+import { IComputedValueOptions } from 'mobx'
+import { onBecomeUnobserved } from 'mobx'
+import { IComputedValue } from 'mobx'
+import { computed } from 'mobx'
+import { PerspectiveCamera } from 'three'
+import * as THREE from 'three'
+import { Scene } from 'three'
+import { Geometry } from 'three'
+import { TextureFilter } from 'three'
+import { Wrapping } from 'three'
+import { Mapping } from 'three'
+import { TextureDataType } from 'three'
+import { PixelFormat } from 'three'
+import { DataTexture } from 'three'
+import { MeshPhongMaterial } from 'three'
+import { MeshBasicMaterial } from 'three'
+import { Color } from 'three'
+import { Object3D } from 'three'
+import { Mesh } from 'three'
+import { Material } from 'three'
+import { ShaderMaterial } from 'three'
+
+import { Vector3 } from '../../math/vector3'
+
+type ComputedFn<T> = (func: () => T, options?: IComputedValueOptions<T>) => IComputedValue<T>
+
+const computedInstance = <T, O>(
+  create: (opts: O) => T,
+  update: (instance: T, opts: O) => void,
+  computedFn: ComputedFn<T> = computed,
+) => (expr: () => O): IComputedValue<T> => {
+  let instance: T
+  return computedFn(() => {
+    const opts = expr()
+    if (instance == null) {
+      instance = create(opts)
+    }
+    update(instance, opts)
+    return instance
+  }, { equals: () => false })
```

Comment:
```ts
{ equals: () => false }
```

This is the magic I was missing. If I thought about this earlier I wouldn't have bothered with the reconciler 😄 

This allows consumers to recompute even if we return the same instance.

Should this ever return true? Probably not.

If this `equals` method is ever called, it means an observable dependency has changed, which means a 99% chance that we no longer equal. So there is little to zero reason to waste CPU cycles confirming that we don't. It's a win-win!

Reply:
The only reason I would still thing equals true might be useful is if you have a dependency in your dependency that changed, and you yourself didn't change. In which case... does always returning equals true work too?